### PR TITLE
Sequence support

### PR DESCRIPTION
**Pull request by iceTTTT**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #6590
 
support Create/Drop/Show sequence.
support nextval/setval/currval/lastval func for sequence.

https://github.com/iceTTTT/docs/blob/sequence/design/sequence.md   design doc

## What this PR does / why we need it:

### CODE REVIEW

**Review by ouyuanning — APPROVED** [review]

**Review by aunjgr — APPROVED** [review]

**Review by zhangxu19830126 — APPROVED** [review]

**Review by iamlinjunhong — APPROVED** [review]

File: pkg/frontend/authenticate.go [hunk 1]
```diff
@@ -452,15 +452,19 @@ const (
 	PrivilegeTypeUserOwnership
 	PrivilegeTypeRoleOwnership
 	PrivilegeTypeShowTables
+	PrivilegeTypeShowSequences
```
Comment (daviszhen) [inline pkg/frontend/authenticate.go:455]: 跟张潇 @domingozhang 确认下。有没有show/create/drop/alter sequence的权限。我印象中没有。

**Review by daviszhen — CHANGES_REQUESTED** [review]
需要一个design docs 介绍下怎么实现sequence和相关函数的。

**Comment by iceTTTT** [comment]
https://github.com/iceTTTT/docs/blob/sequence/design/sequence.md  design doc

**Review by reusee — APPROVED** [review]

File: pkg/frontend/authenticate.go (hunk 1 above)
Comment (iceTTTT) [reply pkg/frontend/authenticate.go]: Removed  privilege related code.

File: pkg/sql/plan/function/builtin/seq/seq_util.go [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+	defer cancel()
+	if err := eg.Rollback(ctx, txn); err != nil {
+		return err
+	}
+	err := txn.Rollback(ctx)
+	txn = nil
+	return err
+}
+
+// func makeBatch(values []interface{}, proc *process.Process) (*batch.Batch, error) {
```
Comment (daviszhen) [inline pkg/sql/plan/function/builtin/seq/seq_util.go:91]: just remove these unused code

**Review by daviszhen — APPROVED** [review]

**Review by nnsgmsone — APPROVED** [review]

**Pull request merged**